pandas/core/arrays/arrow/_arrow_utils.py

```diff
@@ -19,9 +19,12 @@ def fallback_performancewarning(version: str | None = None) -> None:
     Raise a PerformanceWarning for falling back to ExtensionArray's
     non-pyarrow method
     """
-    msg = "Falling back on a non-pyarrow code path which may decrease performance."
+    msg = (
+        "Falling back on a non-pyarrow code path which may decrease performance or "
+        "not be fully compatible with pyarrow."
```

Comment:
Is there a specific reason for adding this additional sentence? 
(since it is so generic, it's also not that useful / potentially confusing I think)

Reply:
I also wanted to convey in this message that a user may get incorrect results if falling back.

e.g. A user's code could fall back to using `ExtensionArray._mode/_quantile` which may have some numpy-based assumptions that may fail when passing arrow (or arrow transformed) data.

Reply:
Yeah that's a good point. Okay I'll leave this message alone, and raise a `NotImplementedError` for the associated case (mode/quantile)